docs/user-guide/templates.md

```diff
@@ -0,0 +1,89 @@
+# Templates
+
+Templates are snippets of predefined code that people can use to replace a job definition in a [screwdriver.yaml](./configuration/index.md). A template contains a series of predefined steps along with a selected Docker image.
+
+## Using a template
+
+To use a template, define a `screwdriver.yaml`:
+
+```yaml
+jobs:
+   main:
+      template: template_name@1.3.0
+```
+
+Screwdriver takes the template configuration and plugs it in, so that the `screwdriver.yaml` becomes:
+
+```yaml
+jobs:
+   main:
+        image: node:6
+        steps:
+          - install: npm install
+          - test: npm test
+          - echo: echo $FOO
+        environment:
+           FOO: bar
+        secrets:
+          - NPM_TOKEN
+```
+
+## Creating a template
+
+### Writing a template
+
+To create a template, create a new repo with a `screwdriver-template.yaml` file. The file should contain a name, version, description, maintainer email, and a config with an image and steps.
+
+Example `screwdriver-template.yaml`:
+
+```yaml
+name: template_name
+version: '1.3'
+description: template for testing
+maintainer: foo@bar.com
+config:
+  image: node:6
+  steps:
+    - install: npm install
+    - test: npm test
+    - echo: echo $FOO
+  environment:
+    FOO: bar
+  secrets:
+     - NPM_TOKEN
+```
+
+### Writing a screwdriver.yaml for your template
```

Comment:
Would it be better to maybe say "template repo" instead of "template"? I was a little confused when I was first reading it